usr/src/test/smartos-test/smartos-test.sh

```diff
@@ -300,6 +300,17 @@ EOF
     fi
 }
 
+#
+# *_test_check() functions exist because some illumos tests assume some
+# reality checks, test drivers, or other prerequisites, that we cannot just
+# extract from the tests-tar tarball.
+#
+
+function driver_install_fail {
```

Comment:
If this change ends up being edited, perhaps this should live up near the definition of `fatal` closer to the top of the file? And perhaps it should call `fatal`??? This feels very nitpicky, but I'm trying to be a detailed reviewer.

Reply:
When I move this function I'll also consider if there are existing bail-out facilities I should use.  Thanks.

Reply:
Fixed.